.teamcity/_self/projects/WPComTests.kt

```diff
@@ -836,3 +839,137 @@ private object I18NTests : BuildType({
 	}
 })
 
+object P2E2ETests : BuildType({
+	name = "P2 E2E Tests"
+	description = "Runs end-to-end tests against P2."
+
+	artifactRules = """
+		logs.tgz => logs.tgz
+		screenshots => screenshots
+		trace => trace
+	""".trimIndent()
+
+	vcs {
+		root(Settings.WpCalypso)
+		cleanCheckout = true
+	}
+
+	params {
+		checkbox(
+			name = "env.SAVE_AUTH_COOKIES",
+			value = "false",
+			label = "Save authentication cookies",
+			description = "Login once and reuse auth cookies for all specs",
+			checked = "true",
+			unchecked = "false"
+		)
+	}
+
+	steps {
+		prepareEnvironment()
+
+		bashNodeScript {
+			name = "Execute tests"
+			scriptContent = """
+				shopt -s globstar
+				set -x
+
+				cd test/e2e
+				mkdir temp
+
+				export URL="https://wpcalypso.wordpress.com"
+
+				export NODE_CONFIG_ENV=test
+				export PLAYWRIGHT_BROWSERS_PATH=0
+				export TEAMCITY_VERSION=2021
+				export LOCALE=en
+				export NODE_CONFIG="{\"calypsoBaseURL\":\"${'$'}{URL%/}\"}"
+				export DEBUG=pw:api
+				export HEADLESS=true
+
+				# Decrypt config
+				openssl aes-256-cbc -md sha1 -d -in ./config/encrypted.enc -out ./config/local-test.json -k "%CONFIG_E2E_ENCRYPTION_KEY%"
+
+				# Run the test
+				export TARGET_DEVICE=desktop
+				export LOCALE=en
+				export NODE_CONFIG="{\"calypsoBaseURL\":\"${'$'}{URL%/}\"}"
+				export DEBUG=pw:api
+
+				yarn jest --reporters=jest-teamcity --reporters=default --maxWorkers=%E2E_WORKERS% --group=p2
+			""".trimIndent()
+			dockerImage = "%docker_image_e2e%"
+		}
+		bashNodeScript {
+			name = "Collect results"
+			executionMode = BuildStep.ExecutionMode.RUN_ON_FAILURE
+			scriptContent = """
+				set -x
+
+				mkdir -p screenshots
+				find test/e2e/results -type f -path '*/screenshots/*' -print0 | xargs -r -0 mv -t screenshots
+
+				mkdir -p logs
+				find test/e2e/results -name '*.log' -print0 | xargs -r -0 tar cvfz logs.tgz
+
+				mkdir -p trace
+				find test/e2e/results -name '*.zip' -print0 | xargs -r -0 mv -t trace
+			""".trimIndent()
+			dockerImage = "%docker_image_e2e%"
+		}
+	}
+
+	features {
+		perfmon {
+		}
+		commitStatusPublisher {
+			vcsRootExtId = "${Settings.WpCalypso.id}"
+			publisher = github {
+				githubUrl = "https://api.github.com"
+				authType = personalToken {
+					token = "credentialsJSON:57e22787-e451-48ed-9fea-b9bf30775b36"
+				}
+			}
+		}
+	}
+
+	triggers {
+		vcs {
```

Comment:
I'm running with the assumption that those tests are testing WPCOM features and not Calypso features, let me know if that's not correct.

But if my assumption is true, then we don't need to automatically run them on branch changes.

Reply:
There's been more build config differences than I expected between WPCOM and Calypso. I had been glossing over it, copy-pasting the config but that'll no longer cut it.

wrt to triggers: yes, I think it's okay leaving it to run on a schedule for the time being. If we see more P2 errors on the frontend that aren't getting caught, then perhaps we'd change the trigger to run on every trunk.

Reply:
vcs trigger removed in d9074c6d975711803c45dc482531f935bce9cae3.